server/service/src/document/update_trigger.rs

```diff
@@ -0,0 +1,67 @@
+use std::str::FromStr;
+
+use chrono::NaiveDate;
+use repository::{StorageConnection, Document, NameRowRepository, NameRow, NameType, Gender as NameRowGender};
+
+extern crate serde;
+extern crate schemafy_core;
+extern crate serde_json;
+
+use serde::{Serialize, Deserialize};
+
+use super::document_service::DocumentInsertError;
+
+
+schemafy::schemafy!(
+  "src/document/schemas/patient.json"
+);
+
+/// Callback called when the document has been updated
+pub fn document_updated(con: &StorageConnection, store_id: &str, doc: &Document) -> Result<(), DocumentInsertError> {
+  match doc.r#type.as_str() {
+    "Patient" => {
+      let patient: Patient = serde_json::from_value(doc.data.clone()).map_err(|err| DocumentInsertError::InternalError(format!("Invalid date of birth format: {}", err)))?;
+      let contact = patient.contact_details.get(0);
+      let date_of_birth =  match patient.date_of_birth{
+        Some(date_of_birth ) => Some(NaiveDate::from_str(&date_of_birth).map_err(|err| DocumentInsertError::InternalError(format!("Invalid date of birth format: {}", err)))?),
+        None => None
+      };
+      let address = patient.addresses.get(0);
+      let name_repo = NameRowRepository::new(con);
+      name_repo.upsert_one(&NameRow {
+        id: format!("{}@{}", patient.id, store_id),
+        name: patient.id,
```

Comment:
oh, right. 
in mSupply the `name` is a concatenation of `first_name` and `last_name` - seems odd that the `id` populates the `name` but that's what you talked about on Tuesday, right? that the `name` here is something unique to the patient and different from the person's name

Reply:
Thanks I will create an update and use the default usages. My intention have a patient at multiple stores / sites...
However, I think I have to defer this problem a little bit and get some API for FE ready. I think I also need to create a name store join to make the patient visible at a store...